File: sdks/ts/src/sdk/models/operations/getfileversionlegalholdsid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class GetFileVersionLegalHoldsIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=file_version_legal_hold_id" })
  fileVersionLegalHoldId: string;
}


export class GetFileVersionLegalHoldsIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: GetFileVersionLegalHoldsIdPathParams;
}


export class GetFileVersionLegalHoldsIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  fileVersionLegalHold?: shared.FileVersionLegalHold;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-sdk/pkg/models/operations/getfileversionlegalholdsid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetFileVersionLegalHoldsIDPathParams struct {
	FileVersionLegalHoldID string `pathParam:"style=simple,explode=false,name=file_version_legal_hold_id"`
}

type GetFileVersionLegalHoldsIDRequest struct {
	Retries    *utils.RetryConfig
	PathParams GetFileVersionLegalHoldsIDPathParams
}

type GetFileVersionLegalHoldsIDResponse struct {
	ClientError          *shared.ClientError
	ContentType          string
	FileVersionLegalHold *shared.FileVersionLegalHold
	StatusCode           int64
}
